pkg/ec2helper/ec2helper.go

```diff
@@ -352,76 +353,137 @@ func (h *EC2Helper) getInstanceTypes(input *ec2.DescribeInstanceTypesInput) ([]*
+		"instance-store": "ubuntu/images/*",
 	},
-	// 64 bit Microsoft Windows Server with Desktop Experience Locale English AMI
 	"Windows": {
-		"ebs": "Windows_Server-????-English-Full-Base-????.??.??",
+		"ebs": "Windows_Server-*-English-Full-Base*",
+	},
+}
+
+// Define all OS and corresponding AMI public parameters path in Parameter Store
+var osSsmPath = map[string]map[string]string{
+	"Amazon Linux": {
+		"ebs":            "/aws/service/ami-amazon-linux-latest",
+		"instance-store": "/aws/service/ami-amazon-linux-latest",
+	},
+	"Amazon Linux 2": {
+		"ebs": "/aws/service/ami-amazon-linux-latest",
+	},
+	"Red Hat": {
+		"ebs": "",
+	},
+	"SUSE Linux": {
+		"ebs": "/aws/service/suse/sles",
+	},
+	"Ubuntu": {
+		"ebs":            "/aws/service/canonical/ubuntu/server/24.04/stable/current",
+		"instance-store": "/aws/service/canonical/ubuntu/server/24.04/stable/current",
+	},
+	"Windows": {
+		"ebs": "/aws/service/ami-windows-latest",
 	},
 }
 
 // Get the appropriate input for describing images
-func getDescribeImagesInputs(rootDeviceType string, architectures []*string) *map[string]ec2.DescribeImagesInput {
+func (h *EC2Helper) getDescribeImagesInputs(rootDeviceType string, architectures []*string) *map[string]ec2.DescribeImagesInput {
+	ssmClient := ssm.New(h.Sess)
+
 	// Construct all the inputs
 	imageInputs := map[string]ec2.DescribeImagesInput{}
 	for osName, rootDeviceTypes := range osDescs {
 
 		// Only add inputs if the corresponding root device type is applicable for the specified os
 		desc, found := rootDeviceTypes[rootDeviceType]
-		if found {
-			imageInputs[osName] = ec2.DescribeImagesInput{
-				Filters: []*ec2.Filter{
-					{
-						Name: aws.String("name"),
-						Values: []*string{
-							aws.String(desc),
-						},
-					},
-					{
-						Name: aws.String("state"),
-						Values: []*string{
-							aws.String("available"),
-						},
+		if !found {
+			continue
+		}
+		imageInputs[osName] = ec2.DescribeImagesInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String("name"),
+					Values: []*string{
+						aws.String(desc),
 					},
-					{
-						Name: aws.String("root-device-type"),
-						Values: []*string{
-							aws.String(rootDeviceType),
-						},
+				},
+				{
+					Name: aws.String("state"),
+					Values: []*string{
+						aws.String("available"),
 					},
-					{
-						Name:   aws.String("architecture"),
-						Values: architectures,
+				},
+				{
+					Name: aws.String("root-device-type"),
+					Values: []*string{
+						aws.String(rootDeviceType),
 					},
-					{
-						Name: aws.String("owner-alias"),
-						Values: []*string{
-							aws.String("amazon"),
-						},
+				},
+				{
+					Name:   aws.String("architecture"),
+					Values: architectures,
+				},
+				{
+					Name: aws.String("owner-alias"),
+					Values: []*string{
+						aws.String("amazon"),
 					},
 				},
-			}
+			},
+		}
+		ssmPath, found := osSsmPath[osName][rootDeviceType]
+		if !found || ssmPath == "" {
+			continue
```

Comment:
what will it happen if we continue if path not found? what default images will it show in the terminal?

Reply:
If no path found like "Red Hat", then will create describeImagesInput base on "name" and other filtering conditions without passing in imageIds we got from SSM path. Basically just behavior like original approach.